src/hooks/useConstructor.ts

```diff
@@ -0,0 +1,32 @@
+import { MutableRefObject, useRef } from "react";
+
+/**
+ * Returns a reference to a constructed value that will not be re-computed
+ * between renders.
```

Comment:
For a React novice like me, a nice additional comment would be "why would I want to use this hook?".  If the important points are that (1) it's not re-computed between renders, and (2) it guarantees non-null, why would I want those features?